File: src/publisher_subscriber/src/py/ps_talker.py
#!/usr/bin/env python
import rospy
from rospy.exceptions import ROSInterruptException

# Importing String message from std_msgs package
from std_msgs.msg import String

# Define the actions of the node
def node():
	# Initialize the node
	# - Name of the node
	# - Whether to make the node different from each other
	# - If anonymous is true then a random number is appended to the name of the node for every
	#   instance of the node created using the rosrun command
	rospy.init_node("ps_talker", anonymous=False)

	# Create a publisher
	# - Name of the topic
	# - Type of the topic
	# - Buffer size to hold messages
	pub = rospy.Publisher('/my_message', String, queue_size = 10)

	# Rate of publication
	rate = rospy.Rate(1) # 1 Hz

	# Start publishing messages
	i = 0
	while not rospy.is_shutdown():
		message = f"hello world {i}"  # Create the message
		rospy.loginfo(message)        # Log the message to stdout
		pub.publish(message)          # Publish the message
		rate.sleep()                  # Sleep for a while so that desired rate of sending is met
		i += 1                        # Increment the counter

# Start the node when the file is executed
if __name__ == "__main__":
	try:
		node()
	except ROSInterruptException:
		pass

File: src/publisher_subscriber/src/cpp/ps_talker.cpp
#include <ros/ros.h>
#include <sstream>

// Importing String message from std_msgs package
#include "std_msgs/String.h"

// Define actions of the node
int main(int argc, char* argv[]) {
  // Initialize the node
  // - argc and argv
  // - Name of the node
  ros::init(argc, argv, "ps_talker");

  // Create a publisher
  // - NodeHandle node
  // - Type of the topic as template argument
  // - Name of the topic
  // - Buffer size to hold messages
  ros::NodeHandle node;
  ros::Publisher pub = node.advertise<std_msgs::String>("/my_message", 10);

  // Rate of publication
  ros::Rate rate(1); // 1 Hz

  // Start publishing messages
  int i = 0;
  while (ros::ok()) {
    std_msgs::String message;               // Create the message
    std::stringstream ss;
    ss << "Hello world " << i;
    message.data = ss.str();
    ROS_INFO("%s", message.data.c_str());   // Log the message to stdout
    pub.publish(message);                   // Publish the message
    ros::spinOnce();                        // Wait for the message to be published
    rate.sleep();                           // Sleep for a while so the desired rate of sending is met
    ++i;                                    // Increment the counter
  }

  return 0;
}
